src/bitmessagekivy/baseclass/draft.py

```diff
@@ -0,0 +1,203 @@
+        draftCnt_obj.ids.badge_txt.text = showLimitedCnt(int(Count))
+
+    def set_mdList(self):
+        """This method is used to create mdlist"""
+        data = []
+        total_draft_msg = len(self.ids.ml.children)
+        for mail in self.queryreturn:
+            third_text = mail[3].replace('\n', ' ')
+            data.append({
+                'text': mail[1].strip(),
+                'secondary_text': mail[2][:10] + '...........' if len(
+                    mail[2]) > 10 else mail[2] + '\n' + " " + (
+                        third_text[:25] + '...!') if len(
+                            third_text) > 25 else third_text,
+                'ackdata': mail[5], 'senttime': mail[6]})
+        for item in data:
+            message_row = SwipeToDeleteItem(
+                text='Draft',
+            )
+            listItem = message_row.ids.content
+            listItem.secondary_text = item["text"]
+            listItem.theme_text_color = "Custom"
+            listItem.text_color = ThemeClsColor
+            message_row.ids.avater_img.source = os.path.join(state.imageDir, 'avatar.png')
+            listItem.bind(on_release=partial(
+                self.draft_detail, item['ackdata'], message_row))
+            message_row.ids.time_tag.text = str(ShowTimeHistoy(item['senttime']))
+            message_row.ids.delete_msg.bind(on_press=partial(self.delete_draft, item['ackdata']))
+            self.ids.ml.add_widget(message_row)
+        updated_msg = len(self.ids.ml.children)
+        self.has_refreshed = True if total_draft_msg != updated_msg else False
+
+    def check_scroll_y(self, instance, somethingelse):
+        """Load data on scroll"""
+        if self.ids.scroll_y.scroll_y <= -0.0 and self.has_refreshed:
+            self.ids.scroll_y.scroll_y = 0.06
+            total_draft_msg = len(self.ids.ml.children)
+            self.update_draft_screen_on_scroll(total_draft_msg)
+
+    def update_draft_screen_on_scroll(self, total_draft_msg, where='', what=''):
+        """Load more data on scroll down"""
+        self.draftDataQuery('fromaddress', where, what, total_draft_msg, 5)
+        self.set_mdList()
+
+    def draft_detail(self, ackdata, instance, *args):
+        """Show draft Details"""
+        if instance.state == 'closed':
+            instance.ids.delete_msg.disabled = True
+            if instance.open_progress == 0.0:
+                state.detailPageType = 'draft'
+                state.mail_id = ackdata
+                if self.manager:
+                    src_mng_obj = self.manager
+                else:
+                    src_mng_obj = self.parent.parent
+                src_mng_obj.screens[11].clear_widgets()
+                src_mng_obj.screens[11].add_widget(MailDetail())
+                src_mng_obj.current = 'mailDetail'
+        else:
+            instance.ids.delete_msg.disabled = False
+
+    def delete_draft(self, data_index, instance, *args):
+        """Delete draft message permanently"""
```

Comment:
the backend code from this should be extracted and put into for example `helper_sent.py / delete`